ls1intum/Artemis
# Update the courses management overview

### Checklist

### Description

As discussed in meetings and on Slack when the mock-up was shared. The new overview has tiles for each course, and shows exercises split into the 3 categories "future" (not released), "current" (not due) and "past" (assessment in progress or due). Each tile also features statistics, which at the moment only include the active students, but will be expanded in the future. (E.g. with course overall score.)

### Steps for Testing

1. View the Course Management page
2. Check if everything displays correctly (e.g. no "null" or "undefined" strings in the statistics)
3. Check if the links and buttons work as intended
4. Check that as far as you can tell, that the progress bars on the left side of the exercise-specific buttons are correct
4. Make sure there is no big performance impact when using the page only for transit (i.e. the courses should load fast, while the exercises and stats can take a bit longer, allowing to access the courses fast without waiting for the other things to load)
5. The guided tour should still work.

We also suggest to test the sorting and stats of exercises:
1. Create an exercise with the release date in the future. It should be in the "Released Soon" group if the release date is within the next seven days, otherwise it should not be displayed.
2. Release the exercise it should now be in "Currently In Progress". Add a due date in the future to the exercise and it should still be in that group. While in this group the amount of participations divided by the total number of students is shown as statistic in the progress bar (in the exercise list).
3. Set an assessment due date in the future. Once the due date passes the exercise appears in "Currently In Assessment" (if **no** assessment due date is set at this point, the exercise shows in the "Past Exercises" group).
4. While in "Currently In Assessment" the number of assessments divided by the number of submissions is displayed.
5. Once the assessment due date passes the exercise is in "Past Exercises" where the average score is shown.

After participating in an exercise the "Active Students" graph should resemble that.

### Test Coverage

course-management-overview-exercise-row: 100%
course-management-overview-card: 100%
course-management-overview-statistics: 100%
course-management: 86.59%
CourseResource: 91%
CourseService: 94%
ExerciseService: 89%

### Screenshots

Every Course is displayed as one tile. Categories are listed next to the title. Exercises are grouped into "Released Soon" (release in the next 7 days), "Currently In Progress" (released, but not due), "Currently In Assessment" (due for students but not assessment due) and "Past Exercises" (assessment was over in the last 7 days). Hidden quiz exercises are not show, but visible ones are in "Released Soon" until they get set to active and move to the "Currently In Progress" group. The tiles are arranged among each other. A tile itself looks like this:



![Screenshot 2021-03-22 at 14 34 31](https://user-images.githubusercontent.com/44401560/111998440-481ae780-8b1c-11eb-85b1-885c392195cd.png)


Tooltips are present for all buttons to be able to tell them apart.

Different groups (Released soon, Currently In Progress, etc.) provide different kinds of information shown in the progress bar. " Currently In Progress" for example, shows the current amount of participations:

![Screenshot 2021-03-22 at 14 42 21](https://user-images.githubusercontent.com/44401560/111999025-ddb67700-8b1c-11eb-808c-41b224a73076.png)



Courses can also have icons, which are displayed as well (see top left):

![Screenshot 2021-03-22 at 14 26 24](https://user-images.githubusercontent.com/44401560/111997019-e4dc8580-8b1a-11eb-9593-270b25f44fa6.png)

When resizing the window, the tiles act accordingly: First, the categories and the progress bar disappears. Step by step less and less information is shown (Compare screenshot 1 and following):



![Screenshot 2021-03-22 at 14 39 27](https://user-images.githubusercontent.com/44401560/111998606-7993b300-8b1c-11eb-964e-5d183d01efd1.png)
